app/src/main/java/com/google/android/sambadocumentsprovider/browsing/NetworkBrowser.java

```diff
@@ -58,7 +52,19 @@ public AsyncTask getServersAsync(OnTaskFinishedCallback<List<String>> callback)
   }
 
   private List<String> getServers() throws BrowsingException {
-    return mMasterProvider.getServers();
+    List<String> servers = null;
+
+    try {
+      servers = mMasterProvider.getServers();
+    } catch (BrowsingException e) {
```

Comment:
It will throw an exception if there're no master browsers on the network. That's why below I'm checking for either an exception or an empty list.